dicodile/update_z/tests/test_distributed_sparse_encoder.py

```diff
@@ -25,31 +26,63 @@ def test_distributed_sparse_encoder():
                   freeze_support=False, warm_start=False, random_state=27)
 
     X = rng.randn(n_channels, n_times)
-    D = rng.randn(n_atoms, n_channels, n_times_atom)
-    sum_axis = tuple(range(1, D.ndim))
-    D /= np.sqrt(np.sum(D * D, axis=sum_axis, keepdims=True))
-    DtD = compute_DtD(D)
+    if not rank1:
+        D = rng.randn(n_atoms, n_channels, *atom_support)
+        sum_axis = tuple(range(1, D.ndim))
+        D /= np.sqrt(np.sum(D * D, axis=sum_axis, keepdims=True))
+    else:
+        u = rng.randn(n_atoms, n_channels)
+        v = rng.randn(n_atoms, *atom_support)
+        # XXX normalize?
```

Comment:
Yes

Reply:
OK, but I'm not sure how to normalize rank-1 dicts :-)

Reply:
you can simply normalize `u` and `v` separately.
Note that this is in essence what is done in `prox_uv`, which is a projection on the unit ball with `uv_constraint` either `joint` (normalize both at the same time) or `separate` (project on each dimension).